tests/appsec/iast/source/test_iast_request_cookie.py

```diff
@@ -0,0 +1,31 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+import pytest
+import re
+from utils import weblog, interfaces, context, coverage, released, missing_feature
+
+
+if context.library == "cpp":
+    pytestmark = pytest.mark.skip("not relevant")
+
+
+@coverage.basic
+@released(dotnet="?", golang="?", php_appsec="?", python="?", ruby="?")
+@released(java={"spring-boot": "1.5.0", "spring-boot-jetty": "1.5.0", "spring-boot-openliberty": "1.5.0", "*": "?"})
+@released(nodejs="?")
+class TestRequestCookie:
+    """Verify that request cookies are tainted"""
+
+    def test_cookie(self):
+        weblog.get("/iast/source/cookie/test", cookies={"cookie-source-name": "cookie-source-value"})
+        interfaces.library_stdout.wait()
+        test_name = re.escape(
+            'tainted={"value":"cookie-source-name","ranges":[{"source":{"origin":"http.request.cookie.name","name":"cookie-source-name","value":"cookie-source-name"},"start":0,"length":18}]}'
+        )
+        interfaces.library_stdout.assert_presence(test_name, level="DEBUG")
```

Comment:
Actually testing this with logs forces every tracer to have this logging mechanism, and have it with the same format. I wonder if it's better to detect it using simple injection vulnerability?

I'm fine with merging this PR as it stands. But we might need to change this approach later to make it easier for other languages.

Reply:
Let me try, I think it could be done in a few hours 🙏  I have to: 

- Extend the iast system test functionality to deal with sources and not only with vulnerabilities 
- Add vulnerable code to each controller endpoint without reusing code to avoid the Overhead controller's filter